sabnzbd/newsunpack.py

```diff
@@ -2077,6 +2077,18 @@ def quick_check_set(set, nzo):
             if nzf.md5sum == md5pack[file]:
                 try:
                     logging.debug("Quick-check will rename %s to %s", nzf.filename, file)
+
+                    # Let's check if there is a subdir in 'file', and if so, pre-create it
+                    new_path = re.search("(.*)/", file)
+                    if new_path:
+                        new_path = os.path.join(nzo.download_path, new_path.group(1))
```

Comment:
And what if this is a malicious file and they add `../../` stuff?
There should be a check for that.